agent/main.py

```
@@ -234,46 +454,78 @@ def check_availability(vehicle_id: str, pickup_date: str, return_date: str) -> d
+                    existing_re = _parse_date(obj.get('returndate', ''))
+                    if existing_pu and existing_re:
+                        if _dates_overlap(pu, re_d, existing_pu, existing_re):
+                            conflicts.append({
+                                'type': 'booking',
+                                'existing_booking': obj.get('bookingid', ''),
+                                'pickup': obj.get('pickupdate'),
+                                'return': obj.get('returndate'),
+                                'customer': obj.get('custname', ''),
+                                'status': obj.get('bookingstatus', 'Confirmed'),
+                            })
+    except Exception as e:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)